Add an automatic algorithm to find group of triggers, i.e. "takes" or "runs", and attempt splitting recording automagically.

smoia opened the pull request:
<!-- Write all of the issues that are linked to this pull request. -->
<!-- If this PR is enough to close them you can write something like "Closes #314 and closes #42" -->
<!-- If you just want to reference them without closing them, you can add something like "References #112" -->
Closes #

<!-- Add a short description of the PR content here-->


## Proposed Changes
<!-- List major points of changes here, so that the reviewers can have a bit more context while looking at your work! -->
  - Add a brand new cluster estimation to find groups of triggers, i.e. takes or runs.
  - Actually allow SMR (Spike2) files to be processed by phys2bids.
  - Also update workflows since I'm there.

## Change Type
<!-- Indicate the type of change you think your pull request is -->
- [ ] `bugfix` (+0.0.1)
- [x] `minor` (+0.1.0)
- [ ] `major`  (+1.0.0)
- [ ] `refactoring` (no version update)
- [ ] `test` (no version update)
- [x] `infrastructure` (no version update)
- [ ] `documentation` (no version update)
- [ ] `other`

## Checklist before review
<!-- If this section is not clear, please read this part of the docs: https://phys2bids.readthedocs.io/en/latest/contributorfile.html#pr -->
<!-- You're invited to open a draft PR ASAP, but before marking it "ready for review", check that you have done the following: -->
- [x] I added everything I wanted to add to this PR.
- [x] \[Code or tests only\] I wrote/updated the necessary docstrings.
- [ ] \[Code or tests only\] I ran and passed tests locally.
- [ ] \[Documentation only\] I built the docs locally.
- [x] My contribution is harmonious with the rest of the code: I'm not introducing repetitions.
- [x] My code respects the adopted style, especially linting conventions.
- [x] The title of this PR is explanatory on its own, enough to be understood as part of a changelog.
- [x] I added or indicated the right labels.
<!-- If relevant, you can add a milestone label or indicate an ideal timeline for this PR, including whether the progress of this PR is linked to other PRs being completed before or after it. -->
- [ ] I added information regarding the timeline of completion for this PR.
<!-- If you want, you can ask for reviews on a draft PR -->
- [ ] Please, comment on my PR while it's a draft and give me feedback on the development!

smoia committed:
Actually allow SMR files to be processed

smoia committed:
Update workflows

smoia committed:
Fix some string issues

smoia committed:
Update setup to newer physiopy

smoia committed:
Add a brand new cluster estimation to find groups of triggers, i.e. takes or runs.

smoia committed:
numpy docstrings.

smoia committed:
Fix issue with help

smoia committed:
Fix more explanation of CI and also add a tiny comment to stop puzzling about why it works.

RayStick commented on phys2bids/cli/run.py:
```diff
@@ -134,6 +134,32 @@ def _get_parser():
+        help="Run automatic algorithm to estimate clusters of triggers, i.e. the "
+        "'takes' or 'runs' of fMRI. Useful when sequences were stopped and restarted, "
+        "or when you don't know how many triggers or trs you have in each take. "
+        "This might work 95%% of the time. Default is False.",
```
"This might work 95%% of the time" is a funny sentence! Can we be clearer on why it may not? Or can we say something like "please check this output/plot carefully to see if it has done X task correctly"

smoia committed:
Update phys2bids/slice4phys.py

Co-authored-by: Rachael Stickland <50215726+RayStick@users.noreply.github.com>

smoia committed:
Merge remote-tracking branch 'origin/master' into enh/spikegroups

smoia committed:
Remove redundant code

smoia committed:
Address review comments and add test

smoia committed:
Add two break tests

smoia committed:
Fix imports

smoia committed:
CircleCI stop being a d*

smoia commented:
@RayStick I tried to incorporate your suggestion and added a few tests to check the new function (both unit and break tests). I also added some coverage check skips in places where we can't check them (for any reason whatsoever)
